### PR TITLE
Fix: IOU description can't be seen on the LHN reports

### PR DESCRIPTION
**Pull request by tobyjsullivan**
### Details

Update the translation `iou.payerOwesAmount` to include the IOU description when it is available. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/31859
PROPOSAL: https://github.com/Expensify/App/issues/31859#issuecomment-1826133247

### Tests

**Steps to Reproduce**
1. Navigate to `https://dev.new.expensify.com:8082/`
2. Click avatar at top of LHN to open "Settings"
3. Click "Preferences"
4. Set "Priority mode" to "Most recent"
5. Click the FAB to open menu.
6. Click "Request Money".
7. Select "Manual" entry.
8. Enter amount of `5000` and click "Next".
9. Enter email for another member and click the suggestion.
10. Enter a description (e.g., "Ramen dinner")
11. Click "Request $5000.00" to create the IOU
12. Find the IOU report in LHN for one of the members.
13. Verify the description is included in the report preview in the LHN.

**Tests**
1. Size window to full-size. Observe the IOU description is included in the LHN on desktop view.
   - Note: if the email address is too long, this can be verified by inspecting the element in Chrome dev tools.
2. Resize screen to mobile size, click back arrow to view LHN. Observe description appears on IOU report.
3. Verify that no errors appear in the JS console.
4. Create a new split IOU with no description. Confirm the LHN record displays the amount without a description.
5. Split the IOU between multiple members.
6. Edit the amount of an IOU.

### Offline tests

1. Throttle network (in Chrome dev tools).
2. Ensure LHN still renders correctly.

### QA Steps

**Steps to Reproduce**
1. Navigate to `https://staging.new.expensify.com/`
2. Sign into an account with a workspace
3. Click avatar at top of LHN to open "Settings"
4. Click "Preferences"
5. Set "Priority mode" to "Most recent"
6. Click the FAB to open menu.
7. Click "Request Money".
8. Select "Manual" entry.
9. Enter amount of `5000` and click "Next".
10. Enter email for another member and click the suggestion.
11. Enter a description (e.g., "Ramen dinner")
12. Click "Request $5000.00" to create the IOU
13. Find the IOU report in LHN for one of the members.
15. Verify the report subtitle includes the description.
    - You may need to resize the browser window to mobile size to see the full description (see screenshots below for examples)

**Tests**
- [ ] Size window to full-size. Observe the IOU description is included in the LHN on desktop view.
   - Note: if the email address is too long, this can be verified by inspecting the element in Chrome dev tools.
- [ ] Resize screen to mobile size, click back arrow to view LHN. Observe description appears on IOU report.
- [ ] Verify that no errors appear in the JS console.
- [ ] Create a new IOU with no description. Confirm the LHN record displays the amount without a description.
- [ ] Split the request between multiple members.
- [ ] Edit the amount of the money request and verify the description is updated.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account]
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message: https://expensify.slack.com/archives/C01GTK53T8Q/p1705698092241939
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

**Note:** change is not actually visible on Android. However, tests confirm no functionality is broken.

https://github.com/Expensify/App/assets/1534986/f2c1806f-7f94-4da9-86e6-5b2c39119cd1

</details>

<details>
<summary>Android: mWeb Chrome</summary>

**Note:** change is not actually visible on Android. However, tests confirm no functionality is broken.

https://github.com/Expensify/App/assets/1534986/cc777dae-ed5b-4f41-a2f4-68cc123730ac

</details>

<details>
<summary>iOS: Native</summary>

**Note:** change is not actually visible on iOS. However, tests confirm no functionality is broken.

https://github.com/Expensify/App/assets/1534986/d2ae790a-965a-4c52-832e-dab91fed0e05

</details>

<details>
<summary>iOS: mWeb Safari</summary>

**Note:** change is not actually visible on iOS. However, tests confirm no functionality is broken.

https://github.com/Expensify/App/assets/1534986/63ae4e55-4bb1-4bd0-b533-8417f0674812

</details>

<details open>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/1534986/380983eb-dbfa-4aa2-9b4c-28dd3d122d78

image

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/1534986/5ee0ca47-ae88-40c4-935b-c141664a195c

</details>

### CODE REVIEW

**Comment by tobyjsullivan** [comment]
_**I have read the CLA Document and I hereby sign the CLA**_

**Comment by dangrous** [comment]
hey @parasharrajat! Do you have time to take a look at this today? Thanks!

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/libs/ReportUtils.ts [hunk 1]
```diff
@@ -1931,7 +1931,9 @@ function getMoneyRequestReportName(report: OnyxEntry<Report>, policy: OnyxEntry<
     }
 
     if (isProcessingReport(report) || isDraftExpenseReport(report) || moneyRequestTotal === 0) {
-        return Localize.translateLocal('iou.payerOwesAmount', {payer: payerOrApproverName, amount: formattedAmount});
+        const reportTransactions = !isEmptyObject(report) ? TransactionUtils.getAllReportTransactions(report.reportID) : [];
+        const comment = reportTransactions.length === 1 ? TransactionUtils.getDescription(reportTransactions[0]) : undefined;
+        return Localize.translateLocal('iou.payerOwesAmount', {payer: payerOrApproverName, amount: formattedAmount, comment});
```
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts:1936]: The translation is being updated to include the description from money requests.

File: src/libs/actions/IOU.js [hunk 2]
```diff
@@ -2272,10 +2273,16 @@ function editRegularMoneyRequest(transactionID, transactionThreadReportID, trans
...
+        const payer = ReportUtils.getPersonalDetailsForAccountID(updatedMoneyRequestReport.managerID).login || '';
+        const formattedAmount = CurrencyUtils.convertToDisplayString(updatedMoneyRequestReport.total, updatedMoneyRequestReport.currency);
+        let messageText;
+        if (hasNonReimbursableTransactions) {
+            messageText = Localize.translateLocal('iou.payerSpentAmount', {payer, amount: formattedAmount});
+        } else {
+            const comment = TransactionUtils.getDescription(updatedTransaction);
+            messageText = Localize.translateLocal('iou.payerOwesAmount', {payer, amount: formattedAmount, comment});
+        }
+
```
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/actions/IOU.js:2285]: The previous code used a ternary operation to select which translation is used, then the same parameters were passed in either case.

Now, the `iou.payerOwesAmount` translation is being updated to accept a description from money requests. As a result, the translations no longer accept the same parameters and the previous ternary branch is no longer an option.

File: src/libs/actions/IOU.js [hunk 3]
```diff
@@ -2563,10 +2570,16 @@ function deleteMoneyRequest(transactionID, reportAction, isSingleTransactionView
...
+    const payer = ReportUtils.getPersonalDetailsForAccountID(updatedIOUReport.managerID).login || '';
+    const formattedAmount = CurrencyUtils.convertToDisplayString(updatedIOUReport.total, updatedIOUReport.currency);
+    let messageText;
+    if (hasNonReimbursableTransactions) {
+        messageText = Localize.translateLocal('iou.payerSpentAmount', {payer, amount: formattedAmount});
+    } else {
+        const comment = !isEmptyObject(transaction) ? TransactionUtils.getDescription(transaction) : undefined;
+        messageText = Localize.translateLocal('iou.payerOwesAmount', {payer, amount: formattedAmount, comment});
+    }
+
```
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/actions/IOU.js:2582]: The previous code used a ternary operation to select which translation is used, then the same parameters were passed in either case.

Now, the `iou.payerOwesAmount` translation is being updated to accept a description from money requests. As a result, the translations no longer accept the same parameters and the previous ternary branch is no longer an option.

File: src/libs/ReportUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts:1936]: Okay, thanks for the clarification. Re-reading the discussion on the original issue, I think I misunderstood [this comment](https://github.com/Expensify/App/issues/31859#issuecomment-1887487141).

> Correct, let's change the existing one and adjust it where used.

I originally interpreted this to mean we should include the description everywhere the translation is used. However, now I see that's not what you meant. Instead, you meant we can use the same translation everywhere and other places would not need to change.

I can revert the changes to these blocks. Please confirm if that's the preferred course of action.

File: src/libs/actions/IOU.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/actions/IOU.js:2582]: Okay, thanks for the clarification. Re-reading the discussion on the original issue, I think I misunderstood [this comment](https://github.com/Expensify/App/issues/31859#issuecomment-1887487141).

> Correct, let's change the existing one and adjust it where used.

I originally interpreted this to mean we should include the description everywhere the translation is used. However, now I see that's not what you meant. Instead, you meant we can use the same translation everywhere and other places would not need to change.

I can revert the changes to these blocks. Please confirm if that's the preferred course of action.

File: src/libs/actions/IOU.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/actions/IOU.js:2285]: Okay, thanks for the clarification. Re-reading the discussion on the original issue, I think I misunderstood [this comment](https://github.com/Expensify/App/issues/31859#issuecomment-1887487141).

> Correct, let's change the existing one and adjust it where used.

I originally interpreted this to mean we should include the description everywhere the translation is used. However, now I see that's not what you meant. Instead, you meant we can use the same translation everywhere and other places would not need to change.

I can revert the changes to these blocks. Please confirm if that's the preferred course of action.

File: src/libs/ReportUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts]: Done. I've reverted changes to all the text unrelated to the original issue.

**Comment by tobyjsullivan** [comment]
Updated

**Comment by tobyjsullivan** [comment]
@parasharrajat Okay, good call. I've pushed a commit which adds the comment for that translation.

I can see in the code that there are other cases that probably need a translation (namely, `${name} spent ${amount}`). However, I cannot figure out how to get these cases to show in the UI. Do you know what case would put that in the LHN or should I leave that alone for now?

**Comment by tobyjsullivan** [comment]
Updated

**Comment by tobyjsullivan** [comment]
Updated

**Comment by tobyjsullivan** [comment]
Updated

**Comment by dangrous** [comment]
@parasharrajat do you think you can take another look at this today? Thanks!

**Comment by tobyjsullivan** [comment]
Updated

**Comment by tobyjsullivan** [comment]
Thanks @parasharrajat. Updated

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/libs/ReportUtils.ts [hunk 4]
```diff
@@ -2305,9 +2305,9 @@ function getReportPreviewMessage(
         return reportActionMessage;
     }
 
+    const linkedTransaction = !isEmptyObject(reportAction) ? TransactionUtils.getLinkedTransaction(reportAction) : {};
```
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts]: I've removed this change entirely 👍

File: src/libs/ReportUtils.ts [hunk 5]
```diff
@@ -2341,12 +2341,14 @@ function getReportPreviewMessage(
...
-
-    if (!isEmptyObject(linkedTransaction) && TransactionUtils.hasReceipt(linkedTransaction) && TransactionUtils.isReceiptBeingScanned(linkedTransaction)) {
+    if (
+        !isEmptyObject(linkedTransaction) &&
+        !isEmptyObject(reportAction) &&
+        shouldConsiderScanningReceiptOrPendingRoute &&
+        TransactionUtils.hasReceipt(linkedTransaction) &&
+        TransactionUtils.isReceiptBeingScanned(linkedTransaction) &&
+        ReportActionsUtils.isMoneyRequestAction(reportAction)
+    ) {
```
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts]: That's fair 👍 

I did spend quite a bit of history looking up the (very recent) history of these conditions at convinced myself the conditional loading of `linkedTransaction` is a recently introduced bug. However, I understand completely that fixing this is not worth the risk on this particular PR (assuming I'm even correct in the first place).

As such, I've revisited the changes to this file. The revised version only loads the `linkedTransaction` at the point it is needed to find the comment.

**Comment by tobyjsullivan** [comment]
Updated

**Comment by tobyjsullivan** [comment]
@parasharrajat Thanks for highlighting that.

Please review this video which I recorded off current `main` and shows the same behavior for the amounts. I believe attempting to fix this goes beyond the scope of my original proposal and is not practical to include in this PR.

Does this PR make the current issue worse?

**Demo from `main`** (not this PR)

https://github.com/Expensify/App/assets/1534986/62cd1bf6-b87f-4c79-8799-7b3a23d6bee9

**Comment by tobyjsullivan** [comment]
After further investigation, this behaviour is not a bug at all. The logic is quite intentional that this "alternative" text is supposed to show the text of the last message in the thread.

For reference, see the following lines:
https://github.com/Expensify/App/blob/6cd008e9b49f56a098fa7be3ba81bb7b893e4a68/src/libs/SidebarUtils.ts#L288

https://github.com/Expensify/App/blob/6cd008e9b49f56a098fa7be3ba81bb7b893e4a68/src/libs/SidebarUtils.ts#L357

As further evidence, we can inspect the `aria-label` attribute for that preview text:

```html
<div
  dir="auto" 
  aria-label="Last chat message preview"
  class="css-text-146c3p1 r-maxWidth-dnmrzs r-textOverflow-1udbk01 r-wordWrap-1iln25a"
  style="color: rgb(118, 132, 126); font-size: 13px; text-align: left; font-family: ExpensifyNeue-Regular, &quot;Segoe UI Emoji&quot;; text-decoration-line: none; overflow: hidden; min-height: 20px; line-height: 20px; white-space: pre;">
  741f@example.com owes CA$2,954.83 for Film
</div>
```

image

My conclusion here is that the current behaviour in this PR is correct. Showing anything other than the last message text would be a product change which is well outside the scope of PR.

**Comment by tobyjsullivan** [comment]
Okay, I believe the behaviour in this PR is correct and consistent with the current implementation of the LHN. I do not have any knowledge of product decisions related to this UI; however, the current implementation shows the following is intentional.

First, the "alternative text" on the LHN item is supposed to display the text from the last message in the report. It is _not_ a summary of the entire chat. Source: [here](https://github.com/Expensify/App/blob/6cd008e9b49f56a098fa7be3ba81bb7b893e4a68/src/libs/SidebarUtils.ts#L288) and [here](https://github.com/Expensify/App/blob/6cd008e9b49f56a098fa7be3ba81bb7b893e4a68/src/libs/SidebarUtils.ts#L357).

This is further evidenced by the `aria-label` on the text itself which describes this text as a preview of the last message (`aria-label="Last chat message preview"`).

```html
<div dir="auto" aria-label="Last chat message preview" class="css-text-146c3p1 r-maxWidth-dnmrzs r-textOverflow-1udbk01 r-wordWrap-1iln25a" style="color: rgb(118, 132, 126); font-size: 13px; text-align: left; font-family: ExpensifyNeue-Regular, &quot;Segoe UI Emoji&quot;; text-decoration-line: none; overflow: hidden; min-height: 20px; line-height: 20px; white-space: pre;">requested CA$1,259.00 for Film</div>
```

Second, there is additional logic for the case where the last chat item is a preview of multiple money requests. In that case, the last request is specifically selected for the LHN preview ([source](https://github.com/Expensify/App/blob/ba135b3cd5249d78c0e448c115d216e1f3616a6e/src/libs/OptionsListUtils.ts#L549-L564)).

In conclusion, the current implementation of the LHN appears to be intentionally showing the text from the last message and specifically showing the text of the last money request in cases where there are multiple.

**Comment by tobyjsullivan** [comment]
Updated.

Here are comparison screenshots showing the revised PR against staging for various cases.

With regard to case 4, showing details with multiple requests, the LHN shows the preview text for the most recent message which is consistent with current behaviour on staging. The description is now included in this preview text as requested in [this comment](https://github.com/Expensify/App/pull/34840#issuecomment-1916248508).

<details>
<summary><strong>Case 1: Preview for single request</strong></summary>

### Staging

Screenshot - Preview - single - staging


### Current PR

Screenshot - preview - single - dev

</details>

<details>
<summary><strong>Case 2: Details for a single request</strong></summary>

### Staging

Screenshot - Request - single - staging


### Current PR

Screenshot - request - single - dev


</details>

<details>
<summary><strong>Case 3: Preview of multiple requests</strong></summary>

### Staging

Screenshot - Preview - multi - staging


### Current PR

Screenshot - preview - multi - dev


</details>

<details>
<summary><strong>Case 4: Details for multiple requests</strong></summary>

### Staging

**Note:** the current behaviour on staging is to show the _most recent_ requested amount, not the total.

Screenshot - Request - multi - staging


### Current PR

Screenshot - request - multi - dev


</details>

File: src/libs/ReportUtils.ts [hunk 6]
```diff
@@ -2296,6 +2296,7 @@ function getReportPreviewMessage(
     isPreviewMessageForParentChatReport = false,
     policy: OnyxEntry<Policy> = null,
     isForListPreview = false,
+    originalReportAction: OnyxEntry<ReportAction> | EmptyObject = reportAction,
```
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts:2299]: You are correct that they are very similar but there is one case where the values differ.

In general, `reportAction` **should** be the message for which the preview is being generated. However, a different action is passed for `reportAction` when generating the LHN message for a report preview (a message grouping multiple money requests).

https://github.com/Expensify/App/blob/49cb3b1502856a42f9f4ff5c6afbe7b19712153c/src/libs/OptionsListUtils.ts#L558-L573

I believe this is a bug and should be fixed separately.

To generate the correct preview text for a report preview action, we need a way to access the original preview action so that we can determine the number of child reports in the preview. To solve this, I pass the original as `originalReportAction` while leaving `reportAction` unchanged. (Fixing the value of `reportAction` would blow out the scope of this PR, particularly around QA/validation, so I've decided not to do that.)

**Visual Example**

Here is an example of a "report preview" message. In this case, `reportAction` **should** be the last message. However, in practice, `reportAction` is one of the child money request reports. To correct for this, the actual preview message is passed as `originalReportAction`.

image

[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts]: Okay, done with one tweak. I renamed `reportAction ` to `iouReportAction` as I think this will be less confusing to readers. Hopefully the comments clarify why this name makes sense.

**Comment by tobyjsullivan** [comment]
Updated

File: src/libs/ReportUtils.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (tobyjsullivan) [reply src/libs/ReportUtils.ts]: The previous comment suggested `reportAction` could be either a report preview or an IOU report action so perhaps that comment is incorrect.

> `@param [reportAction] This can be either a report preview action or the IOU action`

I did not want to rename `reportAction` to `previewReportAction` because I that name implies the action is a preview (ie, a report preview). This would be confusing because the value is not a report preview action.

I chose `iouReportAction` because it seemed the report action could only be an IOU Report Action. However, if that's not true and it can also be a normal DM, then I no longer believe `iouReportAction` is the better name.

Do you still consider `previewReportAction` to be the correct choice?

**Comment by tobyjsullivan** [comment]
Updated

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by dangrous — DISMISSED** [review]
Cool, this works for me!

**Comment by dangrous** [comment]
Blargh can you try merging main and see if that fixes those performance tests? thanks!

**Comment by dangrous** [comment]
Bump the above ^ @tobyjsullivan - would like to get this merged this week. Thanks!

**Comment by tobyjsullivan** [comment]
Thanks for the nudge. I'd missed that request. Latest `main` is merged in now.

**Review by dangrous — APPROVED** [review]
LGTM!
